src/config.js

```diff
@@ -0,0 +1,58 @@
+/* @flow */
+import requireUncached from 'require-uncached';
+
+import {createLogger} from './util/logger';
+import {UsageError} from './errors';
+
+const log = createLogger(__filename);
+
+type ApplyConfigToArgvParams = {|
+  argv: Object,
+  configObject: Object,
+  defaultValues: Object,
+|};
+
+export function applyConfigToArgv({
+  argv,
+  configObject,
+  defaultValues = {},
+}: ApplyConfigToArgvParams): Object {
+  const newArgv = {...argv};
+  for (const option in configObject) {
+    // we assume the value was set on the CLI if the default value is
+    // not the same as that on the argv object as there is a very rare chance
+    // this happening
+    const wasValueSetOnCLI = typeof(argv[option]) !== 'undefined' &&
+      (argv[option] !== defaultValues[option]);
+    if (wasValueSetOnCLI) {
+      log.debug(`Favoring CLI: ${option}=${argv[option]} over ` +
+        `configuration: ${option}=${configObject[option]}`);
+      continue;
+    }
+    // We want to include options of types string, boolean and number
+    if (argv[option] === undefined) {
+      log.debug(`Favoring configuration: ${option}=${configObject[option]} ` +
+        `over CLI: ${option}=${argv[option]}`);
```

Comment:
This debug message isn't correct. It should say:
````js
log.debug(`Ignoring configuration: ${option}=${configObject[option]} ` +
  'because this is an unknown option');
````